packages/server/lib/refreshTokens.ts

```diff
@@ -30,28 +30,39 @@ export function refreshTokens(): void {
 export async function exec(): Promise<void> {
     logger.info('[refreshTokens] starting');
 
-    const staleConnections = await connectionService.getOldConnections({ days: 7, limit: 500 });
+    const today = new Date();
+    const lockKey = parseInt(`${today.getFullYear()}${today.getMonth() + 1}${today.getDate()}`);
 
-    logger.info(`[refreshTokens] found ${staleConnections.length} stale connections`);
+    // Lock to prevent multiple instances of this cron job from running at the same time
+    await db.knex.transaction(async (trx) => {
+        const { rows } = await trx.raw<{ rows: { pg_try_advisory_xact_lock: boolean }[] }>(`SELECT pg_try_advisory_xact_lock(?);`, [lockKey]);
+        if (!rows || rows.length <= 0 || !rows[0]!.pg_try_advisory_xact_lock) {
+            logger.info(`[refreshTokens] could not acquire lock, skipping`);
+            return;
+        }
+        const staleConnections = await connectionService.getOldConnections({ days: 1, limit: 250 });
 
-    for (const staleConnection of staleConnections) {
-        const { connection_id, environment_id, provider_config_key, account_id } = staleConnection;
+        logger.info(`[refreshTokens] found ${staleConnections.length} stale connections`);
 
-        if (typeof account_id !== 'number') {
-            logger.error(`[refreshTokens] connectionId: ${connection_id} is missing account_id`);
-            continue;
-        }
+        for (const staleConnection of staleConnections) {
+            const { connection_id, environment_id, provider_config_key, account_id } = staleConnection;
+
+            if (typeof account_id !== 'number') {
+                logger.error(`[refreshTokens] connectionId: ${connection_id} is missing account_id`);
+                continue;
+            }
```

Comment:
How can a connection not have an account id?

Reply:
I was trying to use the `NangoConnection` type which had `account_id` as potentially undefined. I updated the method to have an account_id that is set and remove this check with https://github.com/NangoHQ/nango/pull/2105/commits/8596099ad95ab9925fc7e81af0b9731d99741475